R/call_utils.R

```diff
@@ -163,308 +123,86 @@ call_condition_logical <- function(varname, choice) {
 #'   By default `Sys.timezone()` is used.
 #'
 #' @examples
-#' teal.slice:::call_condition_range_posixct(
+#' \dontrun{
+#' call_condition_range_posixct(
 #'   varname = "datetime",
 #'   range = c(Sys.time(), Sys.time() + 1),
 #'   timezone = "UTC"
 #' )
+#' }
 #' @return a `call`
 #' @keywords internal
 call_condition_range_posixct <- function(varname, range, timezone = Sys.timezone()) {
+  checkmate::assert_string(varname)
   checkmate::assert_posixct(range, len = 2, sorted = TRUE)
   checkmate::assert_string(timezone)
-  varname <- call_check_parse_varname(varname)
-
-  range[1] <- trunc(range[1], units = c("secs"))
-  range[2] <- trunc(range[2] + 1, units = c("secs"))
 
-  range <- format(
-    range,
-    format = "%Y-%m-%d %H:%M:%S",
-    tz = timezone
-  )
-
-  call(
-    "&",
-    call(">=", varname, call("as.POSIXct", range[1], tz = timezone)),
-    call("<", varname, call("as.POSIXct", range[2], tz = timezone))
+  sprintf(
+    "%s >= as.POSIXct(\"%s\", tz = \"%s\") & %s < as.POSIXct(\"%s\", tz = \"%s\")",
+    varname, format(range[1], tz = timezone), timezone,
+    varname, format(range[2] + 1, tz = timezone), timezone
```

Comment:
You can avoid passing the same argument to the sprintf by using indices. Feel free to use escape character or just wrapping text in `''` as in suggestion.
```suggestion
    '%1$s >= as.POSIXct("%2$s", tz = "%4$s") & %s < as.POSIXct("%3$s", tz = "%4$s")',
    varname, format(range[1], tz = timezone), format(range[2] + 1, tz = timezone), timezone
```